File: ruby/bubble_sort.rb
module BubbleSort

	# bubble sort
	def sort(list)
		for i in 0..list.length-2
			for j in 0..list.length-2-i
				if list[j] > list[j+1]
					tmp = list[j]
					list[j] = list[j+1]
					list[j+1] = tmp
				end
			end
		end
	end

  module_function :sort
end

File: java/src/bubblesort/BubbleSort.java
package bubblesort;

import java.util.*;

class BubbleSort {
	 public void Sort(int[] list) {
		 for (int i = 0; i < list.length - 1; i++) {
			for (int j = 0; j < list.length - 1 - i; j++) {
				if (list[j] >list[j + 1]) {
					int tmp = list[j];
					list[j] = list[j + 1];
					list[j + 1] = tmp;
				}
			}
		}
	}
}